### PR TITLE
Opponent’s Results: Inactive event stream, UI, and backend integration

### PR DESCRIPTION
**Pull request by Zymurge**
This PR implements the full Opponent’s Results (formerly passive toasts) feature for the Onion project, covering both backend and frontend, with a focus on reviewable, deduplicated, and user-friendly event history for the inactive player.

**Summary of efforts in this thread:**
- Refined the original passive notification concept into a reviewable, dismissible event stream for the non-active player, now called "Opponent’s Results".
- Updated the project TODO and feature checklist to clarify requirements, event types, and deduplication policy.
- Implemented the event stream contract in the backend and ensured all relevant events (combat, ram, unit destruction, phase changes) are emitted and persisted.
- Extended the frontend transport and session controller to support polling and live event delivery, with proper seq-based deduplication and history loading.
- Added a new right-rail UI panel for "Opponent’s Results" with a compact, scrollable, and dismissible design, showing only new events since last viewed.
- Streamlined the event entry format to a single-line summary per event, removing redundant type and id display for clarity.
- Added and updated focused UI and transport tests to cover all new behaviors, including dismissal, deduplication, and historical event loading.
- Refactored and cleaned up related files, including test plumbing, transport contracts, and the project TODO.
- Moved the entire epic and all subtasks to Done in the TODO, with backend and UI coverage complete.

This PR closes the Opponent’s Results epic and delivers a robust, user-friendly event stream for the inactive player experience.

---

**Please review:**
- UI/UX of the right-rail event stream
- Backend event emission coverage
- Test coverage for edge cases (reconnect, dismissal, history)
- Code structure and maintainability

Ready for review and merge.

### CODE REVIEW

File: web/App.tsx [hunk 1]
```diff
@@ -111,13 +111,26 @@ function App({ gameClient, gameId, liveEventSource, runtimeConfig, showConnectio
...
       }
     }
 
+    if (typeof window !== 'undefined') {
+      console.info('[app] using connected session binding', {
+        hasConnectedSession: connectedSession !== null,
+        connectedGameId: connectedSession?.gameId ?? null,
+      })
+    }
+
```
[the remark this answers is not in the document]
Comment (Zymurge) [reply web/App.tsx:133]: Defered as part of larger logging effort

File: web/components/BattlefieldLeftRail.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                        if (isDestroyed) {
+                          event.stopPropagation()
+                          return
+                        }
+                        event.stopPropagation()
+                        if (activeCombatRole === 'defender') {
+                          onSelectDefenderCombatTarget()
+                        } else {
+                          onSelectUnit(unit.id, event.ctrlKey || event.metaKey)
+                        }
```
[the remark this answers is not in the document]
Comment (Zymurge) [reply web/components/BattlefieldLeftRail.tsx]: fixed

File: web/lib/useInactiveEventStream.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+			return 'Ram resolved'
+		case 'UNIT_STATUS_CHANGED':
+			return 'Unit status changed'
+		case 'PHASE_CHANGED':
+			return 'Phase changed'
+		default:
+			return type.replace(/_/g, ' ').toLowerCase()
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (Zymurge) [reply web/lib/useInactiveEventStream.ts]: Addressed

File: web/lib/useInactiveEventStream.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+				const maxReturnedSeq = events.reduce((maxSeq, event) => Math.max(maxSeq, event.seq), afterSeq)
+				loadedThroughSeqRef.current = Math.max(maxReturnedSeq, lastAppliedEventSeq)
+			} catch {
+				// Keep the existing stream in place; the next live hint can retry.
+			} finally {
+				if (inFlightAfterSeqRef.current === afterSeq) {
+					inFlightAfterSeqRef.current = null
+				}
+			}
```
[the remark this answers is not in the document]
Comment (Zymurge) [reply web/lib/useInactiveEventStream.ts:135]: Addressed

File: web/lib/useInactiveEventStream.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+	}
+
+	if (event.type === 'PHASE_CHANGED') {
+		const from = typeof event.from === 'string' ? event.from : 'unknown'
+		const to = typeof event.to === 'string' ? event.to : 'unknown'
+		const turnNumber = typeof event.turnNumber === 'number' ? ` on turn ${event.turnNumber}` : ''
+		return `Phase changed from ${from} to ${to}${turnNumber}.`
+	}
+
+	return event.type.replace(/_/g, ' ').toLowerCase()
```
[the remark this answers is not in the document]
Comment (Zymurge) [reply web/lib/useInactiveEventStream.ts]: Addressed

File: web/App.tsx [hunk 6]
```diff
@@ -353,339 +373,72 @@ function App({ gameClient, gameId, liveEventSource, runtimeConfig, showConnectio
...
+          selectedCombatAttackStrength={selectedCombatAttackStrength}
+          selectedCombatTarget={selectedCombatTarget}
+          selectedCombatTargetId={selectedCombatTargetId}
+          selectedInspectorDefender={selectedInspectorDefender}
+          selectedInspectorOnion={selectedInspectorOnion}
+          inactiveEventStream={inactiveEventStream}
+          combatTargetOptions={combatTargetOptions}
+          onConfirmCombat={handleConfirmCombat}
+          onSelectCombatTarget={setSelectedCombatTargetId}
+        />
```
[the remark this answers is not in the document]
Comment (Zymurge) [reply web/App.tsx:441]: Might be impacted by subsequent work so skipping

File: web/lib/useConnectionGate.ts [hunk 7]
```diff
@@ -62,6 +70,13 @@ export function useConnectionGate({ runtimeConfig, onConnectedSession }: UseConn
       }
 
       const baseUrl = draft.apiBaseUrl.trim()
+      if (typeof window !== 'undefined') {
+        console.info('[connection-gate] connected session created', {
+          gameId: parsedGameId,
+          baseUrl,
+          username: loginResult.data.userId,
+        })
+      }
```
[the remark this answers is not in the document]
Comment (Zymurge) [reply web/lib/useConnectionGate.ts:79]: Will be addressed in larger effort
